taiga_halo2/src/circuit/vp_circuit.rs

```diff
@@ -6,15 +6,42 @@ use crate::{
     },
     constant::{
         NoteCommitmentDomain, NoteCommitmentFixedBases, NoteCommitmentHashDomain, NUM_NOTE,
+        SETUP_PARAMS_MAP,
     },
     note::Note,
+    vp_description::ValidityPredicateDescription,
 };
+use dyn_clone::{clone_trait_object, DynClone};
 use halo2_gadgets::{ecc::chip::EccChip, sinsemilla::chip::SinsemillaChip};
 use halo2_proofs::{
     circuit::{Layouter, Value},
-    plonk::{ConstraintSystem, Error},
+    plonk::{verify_proof, Circuit, ConstraintSystem, Error, SingleVerifier, VerifyingKey},
+    transcript::Blake2bRead,
 };
-use pasta_curves::pallas;
+use pasta_curves::{pallas, vesta};
+
+#[derive(Debug, Clone)]
+pub struct VPVerifyingInfo {
+    pub param_size: u32,
```

Comment:
we might have to make all VPs the same size at some point - I'm not sure if we need padding or what

Reply:
okay, I think `12` seems good for now.